### PR TITLE
Implement About Button Functionality

### PR DESCRIPTION
**Pull request by BreadFish64**
Fixes the broken "about Citra" button under the help tab.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3005)

### CODE REVIEW

File: src/citra_qt/main.cpp [hunk 1]
```diff
@@ -323,6 +323,11 @@ void GMainWindow::ConnectMenuEvents() {
     ui.action_Show_Filter_Bar->setShortcut(tr("CTRL+F"));
     connect(ui.action_Show_Filter_Bar, &QAction::triggered, this, &GMainWindow::OnToggleFilterBar);
     connect(ui.action_Show_Status_Bar, &QAction::triggered, statusBar(), &QStatusBar::setVisible);
+
+    // Help
+    connect(ui.action_About, &QAction::triggered, []() {
+        QDesktopServices::openUrl(QUrl("https://citra-emu.org/wiki/faq/"));
```
Comment (bunnei) [inline src/citra_qt/main.cpp]: If we're going to make this change, I prefer some form of dialog with text, and a link on it to the FAQ/etc.

**Comment by chris062689** [comment]
For UI changes, please attach screenshots. Thanks!

**Comment by BreadFish64** [comment]
@chris062689 Well it's a little hard to demonstrate this with a static image but here's a [GIF](https://media.giphy.com/media/xT1R9Kghqb63Py39eg/giphy.gif)
@bunnei As for a popup dialog... I could try, are you thinking something like [Dolphin's](https://i.imgur.com/0Ld5Uhr.png)?

**Comment by wwylele** [comment]
I think the FAQ link deserves its own menu item in the "Help" section.

**Comment by BreadFish64** [comment]
Here's the "about Citra" dialog box I came up with
aboutbox
The text was mostly ripped off of dolphin's.
I also linked to the FAQ in the same way as before

File: src/citra_qt/main.cpp [hunk 2]
```diff
@@ -323,6 +323,25 @@ void GMainWindow::ConnectMenuEvents() {
...
+    // Help
+    connect(ui.action_FAQ, &QAction::triggered,
+            []() { QDesktopServices::openUrl(QUrl("https://citra-emu.org/wiki/faq/")); });
+    connect(ui.action_About, &QAction::triggered, this, &GMainWindow::AboutDialog);
+}
+
+void GMainWindow::AboutDialog() {
+    QMessageBox AboutDialog(this);
+    AboutDialog.setText(
+        "Citra is a free and open source 3DS emulator licensed under GPLv2.0.<br/><br/>"
```
Comment (jroweboy) [inline src/citra_qt/main.cpp]: > licensed under GPLv2.0.

licensed under GPLv2.0 or any later version.

**Comment by Schplee** [comment]
I feel that the "About" box should also provide links to the contributors page and full license.

**Comment by BreadFish64** [comment]
aboutbox2
Updated screenshot

**Comment by BreadFish64** [comment]
Travis had a freak accident, my Travis build succeeded

**Comment by Schplee** [comment]
I feel the bullet point style is too cluttersome. 
*And if we're going the [dolphin route](https://github.com/dolphin-emu/dolphin/blob/c004848fd4ee2db0cb9e589e710418d2d34e73c6/Source/Core/DolphinQt2/AboutDialog.cpp), I think you should also include a similar trademark/unaffiliated with Nintendo line*.

**Comment by j-selby** [comment]
Feel like the about box should have "About" in the title, Citra in larger font, version info, compiler info, etc.

**Comment by BreadFish64** [comment]
aboutdialog3
New box, @jroweboy has indicated he wants it to be in a UI file but I'm getting sick of re-inventing the wheel

**Comment by j-selby** [comment]
> "3DS" is a trademarks
trademark

**Comment by BreadFish64** [comment]
image
Fancy pants new dialog box designed in QT Creator

**Comment by RavenHome1** [comment]
maybe leave that anti piracy statement out, does not seem really fitting in there

**Comment by BreadFish64** [comment]
@RavenHome1 I based the text off Dolphin's

**Comment by RavenHome1** [comment]
@BreadFish64 IMO this should tell the user what citra is and leave some useful links, adding some statements on how to use it seems unnecessary

**Comment by BreadFish64** [comment]
> Selby - Yesterday at 7:25 PM
legally own => hold a license for?
Bynari - Yesterday at 7:25 PM
aka merge
Flamboyant Ham - Yesterday at 7:26 PM
@Selby He took dolphin's verbage verbatim lol
Selby - Yesterday at 7:26 PM
ah, kk
¯\_(ツ)_/¯
LGTM
jroweboy - Yesterday at 7:26 PM
its from dolphin? lgtm instamerge kthxbai /s
Breadfish64 - Yesterday at 7:26 PM
the dolphin one says "you have not legally obtained" is that better?(edited)
Flamboyant Ham - Yesterday at 7:27 PM
I think that's better.

File: src/citra_qt/aboutdialog.cpp [hunk 3]
```diff
@@ -10,6 +10,7 @@ AboutDialog::AboutDialog(QWidget* parent) : QDialog(parent), ui(new Ui::AboutDia
     ui->setupUi(this);
     ui->labelBuildInfo->setText(ui->labelBuildInfo->text().arg(
         Common::g_build_name, Common::g_scm_branch, Common::g_scm_desc));
+    setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
```
Comment (jroweboy) [inline src/citra_qt/aboutdialog.cpp]: window flags should be set in the QDialog contrustor http://doc.qt.io/qt-5/qdialog.html#QDialog

Comment (BreadFish64) [reply src/citra_qt/aboutdialog.cpp]: addressed

**Pull request merged**